src/plugins/discover/public/application/main/utils/update_saved_search.ts

```diff
@@ -45,9 +47,12 @@ export function updateSavedSearch(
       .setField('query', services.data.query.queryString.getQuery())
       .setField('filter', services.data.query.filterManager.getFilters());
   } else if (state) {
+    const appFilters = state.filters ? cloneDeep(state.filters) : [];
+    const globalFilters = globalStateContainer.get()?.filters ?? [];
```

Comment:
We use `globalStateContainer` here instead of `filterManager` since `filterManager` is not yet initialized when the page first loads and will return no global filters, but accessing the state directly is reliable.